### PR TITLE
Add fitness curve attribute to Neural Networks

### PR DESCRIPTION
**Pull request by domfrecent**
1) Add Fitness Curve attribute to BaseNeuralNetwork and downstream classes (NN, Linear, Logistic)

2) Fix bug for storing fitness curves in algorithms. Fitness at each iteration should be gathered with get_fitness() rather than get_pop_fitness()

### CODE REVIEW

File: mlrose/neural.py [hunk 1]
```diff
@@ -514,10 +516,10 @@ def fit(self, X, y=None, init_weights=None):
                 if init_weights is None:
                     init_weights = np.random.uniform(-1, 1, num_nodes)
 
-                current_weights, current_loss = random_hill_climb(
+                current_weights, current_loss, curve = random_hill_climb(
```
Comment (gkhayes) [inline mlrose/neural.py]: The number of outputs of random_hill_climb() vary depending on whether curve is True or False, so you're going to need to use an if statement here.

File: mlrose/neural.py [hunk 2]
```diff
@@ -526,21 +528,22 @@ def fit(self, X, y=None, init_weights=None):
         elif self.algorithm == 'simulated_annealing':
             if init_weights is None:
                 init_weights = np.random.uniform(-1, 1, num_nodes)
-            fitted_weights, loss = simulated_annealing(
+            fitted_weights, loss, curve = simulated_annealing(
```
Comment (gkhayes) [inline mlrose/neural.py]: See previous comments.

File: mlrose/neural.py [hunk 3]
```diff
@@ -526,21 +528,22 @@ def fit(self, X, y=None, init_weights=None):
...
                 pop_size=self.pop_size, mutation_prob=self.mutation_prob,
-                max_attempts=self.max_attempts if self.early_stopping else self.max_iters, max_iters=self.max_iters)
+                max_attempts=self.max_attempts if self.early_stopping else self.max_iters,
+                max_iters=self.max_iters, curve=True)
 
         else:  # Gradient descent case
             if init_weights is None:
                 init_weights = np.random.uniform(-1, 1, num_nodes)
-            fitted_weights, loss = gradient_descent(
+            fitted_weights, loss, curve = gradient_descent(
```
[the remark this answers is not in the document]
Comment (domfrecent) [reply mlrose/neural.py]: Added fitness_curve to gradient descent and updated logic here.

File: mlrose/neural.py [hunk 4]
```diff
@@ -514,10 +516,10 @@ def fit(self, X, y=None, init_weights=None):
                 if init_weights is None:
                     init_weights = np.random.uniform(-1, 1, num_nodes)
 
-                current_weights, current_loss = random_hill_climb(
+                current_weights, current_loss, curve = random_hill_climb(
                     problem,
                     max_attempts=self.max_attempts if self.early_stopping else self.max_iters, max_iters=self.max_iters,
-                    restarts=0, init_state=init_weights)
+                    restarts=0, init_state=init_weights, curve=True)
```
[the remark this answers is not in the document]
Comment (domfrecent) [reply mlrose/neural.py]: Good catch, thanks and agreed.

**Comment by domfrecent** [comment]
Thanks for the feedback @gkhayes !

Made some updates to fix the bugs you called out, and also changed the Return variable name from curve -> fitness_curve for distinction between the two attributes.
